app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt

```diff
@@ -0,0 +1,77 @@
+package org.greenstand.android.TreeTracker.messages.announcementmessage
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import app.cash.turbine.test
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.models.messages.MessagesRepo
+import org.greenstand.android.TreeTracker.utils.FakeFileGenerator.fakeAnnouncementMessage
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class AnnouncementViewModelTest{
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val messageId = ""
+    private val messagesRepo = mockk<MessagesRepo>(relaxed = true)
+    private lateinit var announcementViewModel: AnnouncementViewModel
+
+    @Before
+    fun setup(){
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage
+        announcementViewModel = AnnouncementViewModel(messageId, messagesRepo)
+    }
+
+    @Test
+    fun `verify messages repo gets the correct announcement message `()= runBlocking {
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage

```

Comment:
You already put this in the @Before method, so there is no need to put it in all the other methods. Remove this from all other methods.

Reply:
Done